src/components/GameListItem.vue

```diff
@@ -96,6 +114,34 @@ export default {
           this.joiningGame = false;
         });
     },
+    allowDelete() {
+
+      if(this.numPlayers>0 || this.status == true){
+        return false
+
+      }
+      else{
+        return true
+      }
+     
+    },
+    deleteGame() {
+      if(!this.allowDelete()){
+  
+        this.joiningGame = false;
+        this.$store
+        .dispatch('requestDeleteGame', {
+          gameId: this.gameId
+        })
+        .then(() => {
+          this.status = false;
+        })
+        .catch(this.handleError);
+
+      }
+  
+
+    },
```

Comment:
Something like this makes a little more sense imo. Fast fail any errors, make sure to set the joining status immediately and use async/await along with finally to properly set the state even if your call fails or otherwise errors out.
```suggestion
async deleteGame() {
  this.joiningGame = false;

  if (!this.canDeleteGame) {
    // TODO potentially handle error state
    return;
  }

  try {
    await this.$store
      .dispatch('requestDeleteGame', {
        gameId: this.gameId
      });
  } catch (e) {
    this.handleError(e);
  } finally {
    this.status = false;
  }
},
```